libfwupdplugin/fu-security-attr.c

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright (C) 2022 Richard Hughes <richard@hughsie.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#define G_LOG_DOMAIN "FwupdSecurityAttr"
+
+#include "config.h"
+
+#include "fu-security-attr.h"
+
+typedef struct {
+	FuContext *ctx;
+} FuSecurityAttrPrivate;
+
+G_DEFINE_TYPE_WITH_PRIVATE(FuSecurityAttr, fu_security_attr, FWUPD_TYPE_SECURITY_ATTR)
+
+#define GET_PRIVATE(o) (fu_security_attr_get_instance_private(o))
+
+/**
+ * fu_security_attr_add_bios_target_value:
+ * @ctx: a #FuContext
+ * @attr: a #FwupdSecurityAttr
+ * @id: a #FwupdBiosAttr ID or name
+ * @needle: The substring of a target value
+ *
+ * Checks all configured possible values of an enumeration attribute and
+ * if any match @needle then set as the target value.
+ *
+ * Since: 1.8.4
+ **/
+void
+fu_security_attr_add_bios_target_value(FwupdSecurityAttr *attr,
```

Comment:
Maybe make the to_string use the fwupdbiosattr to_string as well for debugging?

Reply:
We don't have the object, we just have the ID string.

Reply:
But you can fetch the object when doing to string. You're saving the context

Reply:
Maybe we want to keep the FwupdBiosAttr ref'd in F**u**SecurityAttr? Will experiment.

Reply:
Hmm, so I did this and it wasn't hugely useful; `fu_util_security_events_to_string()` is the only caller of `fu_security_attr_to_string` and we don't get `FwupdBiosAttr` objects on _events_. What did you have in mind? The Json output?

Reply:
Yeah that's exactly what I had in mind. I was thinking that if you get it as part of the json you get a capture of there being a problem what can fix it and also what the current setting is.

Reply:
I think the only way that's possible is to make fwupd_security_attr_set_bios_attr_id() into fwupd_security_attr_set_bios_attr() -- i.e. pass the full object.

Reply:
I was thinking you can still pass the id but if json or string are requested use the local reference to the context to look it up? But maybe that fails for clients from dbus doesn't it..

Reply:
Ya, it's the wrong layer; no context in libfwupd. I'm not against passing in the object, but it does feel a little circular.

Reply:
It's a weak use case, probably not worth it.

Really long term we should have a full system information to json (hsi, devices, updates, bios attributes). The same stuff that we can use for emulating a system.

So it can just dump as part of that.

Reply:
> So it can just dump as part of that.

I'd forgotten emulation, we probably want that too. Phase 2.

Reply:
Well emulation can already be accomplished from the environment variables and a sysfs tree. But yeah json emulation for everything is preferred.